### ISSUE TITLE
"make check" fails

### ISSUE THREAD

**Issue by ingfra** [dmtx/libdmtx#22]
After building libdmtx 0.7.5 (using "./autogen.sh && ./configure && make") the test-binaries fails to build, eg:
```
$ make check
...
gcc -DHAVE_CONFIG_H -I. -I../..  -Wshadow -Wall -pedantic -ansi   -g -O2 -MT simple_test.o -MD -MP -MF .deps/simple_test.Tpo -c -o simple_test.o simple_test.c
simple_test.c: In function ‘main’:
simple_test.c:41:4: error: C++ style comments are not allowed in ISO C90
    //dmtxEncodeSetProp( enc, DmtxPropPixelPacking, DmtxPack16bppRGB );
    ^
simple_test.c:41:4: error: (this will be reported only once per input file)
simple_test.c:61:32: warning: format ‘%d’ expects argument of type ‘int’, but argument 3 has type ‘size_t {aka long unsigned int}’ [-Wformat=]
    fprintf(stdout, "width:  \"%d\"\n", width);
                                ^
simple_test.c:62:32: warning: format ‘%d’ expects argument of type ‘int’, but argument 3 has type ‘size_t {aka long unsigned int}’ [-Wformat=]
    fprintf(stdout, "height: \"%d\"\n", height);
                                ^
simple_test.c:63:32: warning: format ‘%d’ expects argument of type ‘int’, but argument 3 has type ‘size_t {aka long unsigned int}’ [-Wformat=]
    fprintf(stdout, "bpp:    \"%d\"\n", bytesPerPixel);
                                ^
simple_test.c:65:4: error: ‘for’ loop initial declarations are only allowed in C99 or C11 mode
    for (int i=0; i<width*height; i++){
    ^~~
simple_test.c:65:4: note: use option -std=c99, -std=gnu99, -std=c11 or -std=gnu11 to compile your code
simple_test.c:84:45: warning: format ‘%d’ expects argument of type ‘int’, but argument 3 has type ‘size_t {aka long unsigned int}’ [-Wformat=]
       fprintf(stdout, "msg->arraySize :  \"%d\"\n", msg->arraySize );
                                             ^
simple_test.c:85:45: warning: format ‘%d’ expects argument of type ‘int’, but argument 3 has type ‘size_t {aka long unsigned int}’ [-Wformat=]
       fprintf(stdout, "msg->codeSize  :  \"%d\"\n", msg->codeSize  );
                                             ^
simple_test.c:86:45: warning: format ‘%d’ expects argument of type ‘int’, but argument 3 has type ‘size_t {aka long unsigned int}’ [-Wformat=]
       fprintf(stdout, "msg->outputSize:  \"%d\"\n", msg->outputSize);
                                             ^
simple_test.c:87:18: warning: implicit declaration of function ‘sqrt’ [-Wimplicit-function-declaration]
       int oned = sqrt(msg->arraySize);
                  ^~~~
simple_test.c:87:18: warning: incompatible implicit declaration of built-in function ‘sqrt’
simple_test.c:87:18: note: include ‘<math.h>’ or provide a declaration of ‘sqrt’
simple_test.c:87:7: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
       int oned = sqrt(msg->arraySize);
       ^~~
simple_test.c:88:16: warning: declaration of ‘i’ shadows a previous local [-Wshadow]
       for (int i=0; i<msg->arraySize; i++){
                ^
simple_test.c:65:13: note: shadowed declaration is here
    for (int i=0; i<width*height; i++){
             ^
simple_test.c:88:7: error: ‘for’ loop initial declarations are only allowed in C99 or C11 mode
       for (int i=0; i<msg->arraySize; i++){
       ^~~
simple_test.c:95:7: error: ‘for’ loop initial declarations are only allowed in C99 or C11 mode
       for (int j=0; j<msg->codeSize; j++){
       ^~~
simple_test.c:99:7: error: ‘for’ loop initial declarations are only allowed in C99 or C11 mode
       for (int k=0; k<msg->outputSize; k++){
       ^~~
Makefile:353: recipe for target 'simple_test.o' failed
make[3]: *** [simple_test.o] Error 1
```
Using gcc 6 (on a debian 9):
```
$ gcc --version
gcc (Debian 6.3.0-18+deb9u1) 6.3.0 20170516
```

**Comment by W-Hamra** [comment]
That's because the file is C++ not C, and after compiling it with g++ it'll complain about sqrt missing, because the file doesn't include math.h. So add the include, and manually compile it with g++, it will work.

**Comment by msva** [comment]
Btw, autotools is deprecated, and CMake is recommended way to build the project

**Comment by msva** [comment]
(but still, fix from 129c682 seems to work anyway)

**Comment by msva** [comment]
By the way, isn't it a way for you to upgrade the gcc? I just tested with gcc-9 and clang, and there is no warnings, and all the chain (autogen&&cpnfigure&&make&&make check) finished without any warning
